pkg/segment/query/querystatus.go

```diff
@@ -558,7 +558,20 @@ func GetRemoteRawLogInfo(remoteID string, inrrcs []*utils.RecordResultContainer,
 	}
 	defer arqMapLock.RUnlock()
 
-	return rQuery.searchRes.GetRemoteInfo(remoteID, inrrcs)
+	return rQuery.searchRes.GetRemoteInfo(remoteID, inrrcs, false)
+}
+
+func GetAllRemoteLogs(inrrcs []*utils.RecordResultContainer, qid uint64) ([]map[string]interface{}, []string, error) {
+	arqMapLock.RLock()
+	rQuery, ok := allRunningQueries[qid]
+	if !ok {
+		log.Errorf("GetAllRemoteLogs: qid %+v does not exist!", qid)
+		arqMapLock.RUnlock()
+		return nil, nil, fmt.Errorf("qid does not exist")
+	}
+	defer arqMapLock.RUnlock()
```

Comment:
nit: your unlocking looks correct, but a bit messy. If you only need the lock for `rQuery, ok := allRunningQueries[qid]`, then you can unlock it (not defer unlocking it) right after you read from that map. Or if you need it for longer, you can defer unlocking it right after you lock it

Reply:
oh sorry. Missed it

Reply:
addressed in the latest commit. I have corrected this at a few more places. Later, will go over through this file and see if any more could be cleaned up.